plugins/inputs/pubsub/pubsub.go

```diff
@@ -0,0 +1,290 @@
+package pubsub
+
+import (
+	"cloud.google.com/go/pubsub"
+	"context"
+	"fmt"
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/plugins/inputs"
+	"github.com/influxdata/telegraf/plugins/parsers"
+	"golang.org/x/oauth2/google"
+	"google.golang.org/api/option"
+	"log"
+	"runtime"
+	"sync"
+	"time"
+)
+
+type empty struct{}
+type semaphore chan empty
+
+const (
+	defaultMaxUndeliveredMessages = 1000
+)
+
+type PubSub struct {
+	CredentialsFile string   `toml:"credentials_file"`
+	Project         string   `toml:"google_project"`
+	Subscriptions   []string `toml:"subscriptions"`
+
+	// Subscription ReceiveSettings
+	MaxExtensionSec        int `toml:"max_extension_seconds"`
```

Comment:
If there is a way to TOML a duration, please let me know!

Reply:
You can use `internal.Duration`, here is [an example](https://github.com/influxdata/telegraf/blob/master/plugins/inputs/http/http.go#L30).